datahub_sap_hana/ingestion.py

```diff
@@ -1,40 +1,177 @@
-from typing import Any, Dict
+from collections import defaultdict
+from typing import Any, Dict, Iterable, Union, Tuple, List
 
-import sqlalchemy_hana.types as custom_types  # type: ignore
+import sqlalchemy_hana.types as custom_types 
 from datahub.ingestion.api.common import PipelineContext
 from datahub.ingestion.api.decorators import (
     config_class,  # type: ignore
     platform_name,  # type: ignore
 )
 from datahub.ingestion.source.sql.sql_common import (
-    SQLAlchemySource,
-    register_custom_type,  # type: ignore
+    SQLAlchemySource, SqlWorkUnit,
+    register_custom_type, 
 )
 from datahub.ingestion.source.sql.sql_config import BasicSQLAlchemyConfig
 from datahub.metadata.com.linkedin.pegasus2avro import schema
 
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+
 register_custom_type(custom_types.TINYINT, schema.NumberType)
 
+from pydantic import BaseModel, Field
+from datahub.configuration.common import AllowDenyPattern
+from datahub.emitter import mce_builder
+from datahub.emitter.mcp_builder import mcps_from_mce
+from sqlalchemy import create_engine
+
+#This query only takes 4 columns to match the fields in the ViewLineageEntry class. It also ignores schemas that contain SYS
+# Object dependencies in SAP HANA https://help.sap.com/docs/SAP_HANA_PLATFORM/de2486ee947e43e684d39702027f8a94/5ce9a6584eb84f10afbbf2b133534932.html
+LINEAGE_QUERY = """SELECT BASE_OBJECT_NAME as source_table, BASE_SCHEMA_NAME as source_schema, DEPENDENT_OBJECT_NAME as dependent_view, DEPENDENT_SCHEMA_NAME as dependent_schema
+  from SYS.OBJECT_DEPENDENCIES 
+  WHERE DEPENDENT_OBJECT_TYPE = 'TABLE'
+  OR DEPENDENT_OBJECT_TYPE = 'VIEW'
+  AND BASE_SCHEMA_NAME NOT LIKE '%SYS%'
+  AND DEPENDENT_SCHEMA_NAME NOT LIKE '%SYS%'"""
```

Comment:
Prefer starting on a new line here, for readability

```suggestion
LINEAGE_QUERY = """
SELECT 
  BASE_OBJECT_NAME as source_table, 
  BASE_SCHEMA_NAME as source_schema, 
  DEPENDENT_OBJECT_NAME as dependent_view, 
  DEPENDENT_SCHEMA_NAME as dependent_schema
from SYS.OBJECT_DEPENDENCIES 
WHERE 
  DEPENDENT_OBJECT_TYPE = 'TABLE'
  OR DEPENDENT_OBJECT_TYPE = 'VIEW'
  AND BASE_SCHEMA_NAME NOT LIKE '%SYS%'
  AND DEPENDENT_SCHEMA_NAME NOT LIKE '%SYS%'
"""
```